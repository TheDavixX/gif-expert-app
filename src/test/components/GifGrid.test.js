import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas con <GifGrid />', ()=>{

    const cagetory = "One Punch Man";

    test('debe mostrar <GifGrid /> correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={ cagetory } />);;
        expect( wrapper ).toMatchSnapshot();
    
    });

    test('debe mostrar item cuando se cargan imagenes useFetchGifs', () => {  

        const gifs =[{
            id: 'ABC',
            url: 'htyps://studiandopapu.com',
            title: 'titulo we',
        },
        {
            id: '123',
            url: 'htyps://studiandopapu.com',
            title: 'titulo we',
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ cagetory } />);;
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });

});