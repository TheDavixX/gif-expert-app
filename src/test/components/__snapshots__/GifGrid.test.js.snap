// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifGrid /> debe mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__rubberBand"
  >
    One Punch Man
  </h3>
  <div
    className="card-grid"
  >
    <p
      className="animate__animated animate__rubberBand"
    >
      Loading
    </p>
  </div>
</Fragment>
`;

exports[`Pruebas con <GifGrid /> debe mostrar item cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__rubberBand"
  >
    One Punch Man
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="titulo we"
      url="htyps://studiandopapu.com"
    />
    <GifGridItem
      id="123"
      key="123"
      title="titulo we"
      url="htyps://studiandopapu.com"
    />
  </div>
</Fragment>
`;
